app/github/push_event.rb

```diff
@@ -41,9 +45,14 @@ def deleted?
     event['deleted']
   end
 
-  # The git commit sha to deploy
+  # The git commit sha of this push event.
   def sha
-    event['head_commit']['id']
+    event['head_commit']['id'] unless event['head_commit'].nil?
+  end
+
+  # Returns true if this push event's git commit sha + environment already has an AutoDeployment.
+  def already_has_auto_deployment?(environment)
+    AutoDeployment.exists?(environment: environment, sha: sha)
```

Comment:
I mentioned this in my last review, but there's a more conventional way to do this with activerecord. Essentially, you would add the following the the AutoDeployment model:

```ruby
# Syntax may not be exact, but you get the idea.
validates :sha, uniqueness: { scope: :environment }
```

In turn, creating a duplicate AutoDeployment for the environment will cause the record to be invalid. Right now, that will cause an exception to be raised [here](https://github.com/remind101/slashdeploy/blob/adea526cbfac35f7b001ae2984bc4f382299800e/lib/slashdeploy/service.rb#L37), but we could change that to use `create` (as opposed to `create!`) and handle the `valid?` state.

One possibility would be to DM the user to let them know why the auto deployment wasn't started:

```ruby
unless auto_deployment.valid?
    direct_message \
      auto_deployment.slack_account, \
      ValidationErrorMessage, \
      record: auto_deployment
end
```

Or we can just drop it:

```ruby
return unless auto_deployment.valid?